tests/unit/test_routes/test_utils.py

```diff
@@ -0,0 +1,112 @@
+from app.utils import get_error_code_from_status
+from configs import PaginatorConfig
+
+
+def test_internal_server_error_handler(
+        module_client, module_db, fake_paginated_data_error):
+    client = module_client
+
+    response = client.get('api/v1/resources')
+    assert (response.status_code == 500)
+    assert (isinstance(response.json.get('errors')
+            .get(get_error_code_from_status(response.status_code)), dict))
+    assert (isinstance(response.json.get('errors')
+            .get(get_error_code_from_status(response.status_code)).get('message'), str))
+
+    response = client.get('api/v1/languages')
+    assert (response.status_code == 500)
+    assert (isinstance(response.json.get('errors')
+            .get(get_error_code_from_status(response.status_code)), dict))
+    assert (isinstance(response.json.get('errors')
+            .get(get_error_code_from_status(response.status_code)).get('message'), str))
+
+    response = client.get('api/v1/categories')
+    assert (response.status_code == 500)
+    assert (isinstance(response.json.get('errors')
+            .get(get_error_code_from_status(response.status_code)), dict))
+    assert (isinstance(response.json.get('errors')
+            .get(get_error_code_from_status(response.status_code)).get('message'), str))
+
+
+def test_method_not_allowed_handler(module_client):
+    client = module_client
+
+    response = client.patch('api/v1/resources')
+    assert (response.status_code == 405)
+    assert (isinstance(response.json.get('errors')
+            .get(get_error_code_from_status(response.status_code)), dict))
+    assert (isinstance(response.json.get('errors')
+            .get(get_error_code_from_status(response.status_code)).get('message'), str))
```

Comment:
I think we can DRY this up a lot by having some sort of `validate_error` utility function. We could pass in the response and the status code we're expecting (500, 405, 404, etc) and then just have it check the structure of the error response json and that the status code is correct. We repeat ourselves a lot in these negative tests and it would help us cut down on line count and cognitive complexity to have a helper function for these